File: src/main/java/frc/utils/LEDAnim/RainbowAnim.java
package frc.utils.LEDAnim;

import edu.wpi.first.wpilibj.util.Color;

public class RainbowAnim extends LEDAnim {

    /**
     * Rainbow animation
     * @param length length of LED strip
     * @param speed speed of rainbow, 1 is full cycle per second
     */
    public RainbowAnim(int length){
        super(length);

        for(int i=0; i<leds.length; i++){
            int hue = (int)(((i * 180.0) / leds.length) % 180.0);
            leds[i] = Color.fromHSV(hue, 255, 255);
        }
        initLeds = leds.clone();
    }
    
}

File: src/main/kotlin/frc/utils/LEDAnim/RainbowAnim.kt
package frc.utils.LEDAnim

import edu.wpi.first.wpilibj.util.Color

/**
 * Rainbow animation
 * @param length length of LED strip
 */
class RainbowAnim(length: Int) : LEDAnim(length) {
    init {
        for (i in leds.indices) {
            val hue = (((i * 180.0) / leds.size) % 180.0).toInt()
            leds[i] = Color.fromHSV(hue, 255, 255)
        }
        initLeds = leds.clone()
    }
}
